contributors/design-proposals/daemonset-update.md

```diff
@@ -129,18 +127,35 @@ type DaemonSetStatus struct {
 
 	// UpdatedNumberScheduled is the total number of nodes that are running updated
 	// daemon pod
+	// +optional
 	UpdatedNumberScheduled int32 `json:"updatedNumberScheduled"`
 
 	// NumberAvailable is the number of nodes that should be running the
 	// daemon pod and have one or more of the daemon pod running and
 	// available (ready for at least minReadySeconds)
+	// +optional
 	NumberAvailable int32 `json:"numberAvailable"`
 
 	// NumberUnavailable is the number of nodes that should be running the
 	// daemon pod and have non of the daemon pod running and available
 	// (ready for at least minReadySeconds)
+	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
 }
+
+type DaemonSet struct {
```

Comment:
Could we consider putting the field under the DaemonSetStatus? DeamonSetStatus is generally mutable by the controller, and used to communicate information about the Status of the objects corresponding control loop. My thought is that the current templateGeneration that the DaemonSet controller is attempting to conform its Pods to is part the status of the controller.